lib/plugins/aws/package/compile/events/eventBridge/index.js

```diff
@@ -159,11 +159,20 @@ class AwsCompileEventBridgeEvents {
                     ],
                   ],
                 };
-                iamRoleStatements.push({
+
+                const createDeleteEventBusStmt = {
                   Effect: 'Allow',
                   Resource: eventBusResource,
                   Action: ['events:CreateEventBus', 'events:DeleteEventBus'],
-                });
+                };
+
+                if (
+                  _.findIndex(iamRoleStatements, stmt =>
+                    _.isEqual(stmt, createDeleteEventBusStmt)
```

Comment:
We're trying to avoid lodash for construct which could easily be reflected with native means, and I think here it's the case.

Also with repeated array search we'll not that optimal. Of course in this case it's hard to approach any performance issues, but I believe it'll also be logically cleaner to resort structures as `Map`. Therefore what about switching to logic as:

```javascript
const eventBusIamRoleStatements = new Map();
...
if (!eventBusIamRoleStatements.has(eventBusName)) {
  eventBusIamRoleStatements.set(eventBusName, {
    Effect: 'Allow',
    ...
  });
}
...
iamRoleStatements.push(...eventBusIamRoleStatements.values())
```